### PR TITLE
Optimize template addition & update by cross checking the content with system default templates

### PR DESCRIPTION
**Pull request by darshanasbg**
### Proposed changes in this pull request

$subject. Improves https://github.com/wso2-extensions/identity-event-handler-notification/pull/256.
Related to,
* https://github.com/wso2/product-is/issues/20996

### When should this PR be merged

Immediate.

### Follow up actions

N/A

### CODE REVIEW

File: components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManager.java [hunk 1]
```diff
@@ -260,4 +260,16 @@ private Map<String, Map<String, NotificationTemplate>> getTemplateMap(String not
         }
         return defaultEmailTemplates;
     }
+
+    /**
+     * Checks if the there is a template available as a system default template with the exact same details.
```
Comment (JeethJJ) [inline components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManager.java:265]:
```suggestion
     * Checks if there is a template available as a system default template with the exact same details.
```

**Review by JeethJJ — APPROVED** [review]

**Review by JeethJJ — APPROVED** [review]

**Review by JeethJJ — APPROVED** [review]

File: components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManager.java [hunk 2]
```diff
@@ -39,7 +39,7 @@
  * This class have support for get, list, exists operations and any invocations to modification operations
  * throws {@link UnsupportedOperationException}.
  * This class expected to be use only with conjunction with another {@link TemplatePersistenceManager} implementation
- * which supports full CRUD operations, hence {@link DefaultTemplateManager} provides that aggregation using this as a
+ * which supports full CRUD operations, hence {@link UnifiedTemplateManager} provides that aggregation using this as a
  * fallback provider.
  */
 public class InMemoryBasedTemplateManager implements TemplatePersistenceManager {
```
Comment (madurangasiriwardena) [inline components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManager.java:45]: Renaming this to SystemDefaultTemplateManager would make much sense IMO.

Comment (darshanasbg) [reply components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManager.java:45]: Agreed.. I also thought InMemoryBasedTemplateManager name only gives the unnecessary implementation detail yet not describe the duty of this class.. Will fix this.

Comment (darshanasbg) [reply components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManager.java]: Fixed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/257/commits/8cf3bf4c1b6e6867163740fa63bae3a87a3c43eb

**Review by madurangasiriwardena — APPROVED** [review]

**Pull request merged**